frontend/src/editor/transcription_editor.tsx

```diff
@@ -70,32 +75,54 @@ function renderElement({ element, children, attributes }: RenderElementProps): J
   throw Error('Unknown element type');
 }
 
-function renderLeaf({ leaf, children, attributes }: RenderLeafProps): JSX.Element {
-  const classes = ['word'];
-  if (leaf.start !== undefined) {
-    classes.push(startTimeToClassName(leaf.start));
-  }
+const Leaf = memo(
+  ({
+    start,
+    conf,
+    children,
+    attributes,
+  }: {
+    start?: number;
+    conf?: number;
+    children: RenderLeafProps['children'];
+    attributes: RenderLeafProps['attributes'];
+  }): JSX.Element => {
+    const systemPrefersDark = useMediaQuery('(prefers-color-scheme: dark)');
+    const color = !conf
+      ? undefined
+      : systemPrefersDark
+      ? `hsl(0, 100%, ${Math.min(conf * 50 + 50, 100)}%)`
+      : `hsl(0, 100%, ${Math.min((1 - conf) * 100, 45)}%)`;
+
+    const classes = ['word'];
+    if (start !== undefined) {
+      classes.push(startTimeToClassName(start));
+    }
 
-  const systemPrefersDark = useMediaQuery('(prefers-color-scheme: dark)');
+    return (
+      <span
+        {...attributes}
+        className={classes.join(' ')}
+        onClick={useCallback(() => {
+          // this event is handeled in player.tsx to set the time when someone clicks a word
+          window.dispatchEvent(new TextClickEvent(start));
+        }, [start])}
+        style={{ color }}
+      >
+        {children}
+      </span>
+    );
+  },
+);
 
-  const color = !leaf.conf
-    ? undefined
-    : systemPrefersDark
-    ? `hsl(0, 100%, ${Math.min(leaf.conf * 50 + 50, 100)}%)`
-    : `hsl(0, 100%, ${Math.min((1 - leaf.conf) * 100, 45)}%)`;
+Leaf.displayName = 'Leaf';
 
+function renderLeaf(props: RenderLeafProps): JSX.Element {
```

Comment:
couldn't we just omit this function and pass Leaf directly? If not, cant we just return `<Leaf {...props />` and maybe inline this into the other component?